### PR TITLE
Move files from cirq.google into subpackages

### PR DESCRIPTION
**Pull request by dstrain115**
- Create cirq.google.devices for Google devices
- Create cirq.google.optimizers for gate compilers

### CODE REVIEW

File: cirq/devices/grid_qubit_test.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def test_diagram():
 -----KL-----
 """
     assert (cirq.GridQubit.from_diagram(s) ==
-            cirq.google.known_devices._parse_device(s)[0])
+            cirq.google.devices.known_devices._parse_device(s)[0])
```
Comment (maffoo) [inline cirq/devices/grid_qubit_test.py]: Consider exporting this `_parse_device` function on `cirq.google.devices`. We could rename it to `parse_device_ascii` or something like that.

File: cirq/google/devices/known_devices_test.py [hunk 2]
```diff
@@ -19,11 +19,12 @@ def test_foxtail_qubits():
     for i in range(0, 2):
         for j in range(0, 11):
             expected_qubits.append(cirq.GridQubit(i, j))
-    assert set(expected_qubits) == cirq.google.known_devices.Foxtail.qubits
+    assert set(
+        expected_qubits) == cirq.google.devices.known_devices.Foxtail.qubits
```
Comment (maffoo) [inline cirq/google/devices/known_devices_test.py]: Can just use `cirq.google.Foxtail` here.

File: cirq/protocols/json_test.py [hunk 3]
```diff
@@ -259,10 +259,6 @@ def test_fail_to_resolve():
     cirq.Z,
     'S':
     cirq.S,
-    'SerializableDevice':
-    cirq.google.SerializableDevice.from_proto(
-        proto=cirq.google.known_devices.FOXTAIL_PROTO,
-        gate_set=cirq.google.XMON),
```
Comment (maffoo) [inline cirq/protocols/json_test.py]: Why was this removed?

File: cirq/devices/grid_qubit_test.py (hunk 1 above)
Comment (dstrain115) [reply cirq/devices/grid_qubit_test.py]: I just decided to change this test instead.  I am not sure why it was depending on this function.  It seemed like a bad test.

File: cirq/google/devices/known_devices_test.py (hunk 2 above)
Comment (dstrain115) [reply cirq/google/devices/known_devices_test.py]: Done.

File: cirq/protocols/json_test.py (hunk 3 above)
Comment (dstrain115) [reply cirq/protocols/json_test.py]: Ironically, SerializableDevice cannot yet be serialized into JSON.  I'll add that functionality later.
